dashboards/detailed_info_2.py

```diff
@@ -0,0 +1,132 @@
+import streamlit as st
+
+from utils.duckdb import get_dbcur
+
+def parse_url_query(query_list):
+    # Adapted from process_query_params from auxiliary_functions.py
+    # Functionally does the same but modified slightly due to different structure of arguments passed in
+    # Also returns a json obj instead of py dict so that duckdb recognizes this structure
+    import json
+    result = {}
+    current_field = ""
+
+    for query in query_list:
+        if "=" not in query:
+            continue
+        key, value = query.split("=")
+        if "filters" in key:
+            parts = key.split("[")
+            if len(parts) < 3:
+                continue
+            field_or_value = parts[2].strip("]")
+
+            if field_or_value == "field":
+                # if its a field then use it as a key
+                current_field = value
+                result[current_field] = []
+            elif field_or_value == "type":
+                # there's a type of all in all queries, not sure what that is and whether its relevant
+                continue
+            else:
+                # if its a value then add it to the list by using the last saved field
+                result[current_field].append(value)
+        elif key == "q":
+            result["search_query"] = [value]
+    return json.dumps(result)
```

Comment:
For clarity, I would suggest adding types to this function and indicating the return type.

Also, it would be better for the function to return a Python dictionary instead (so that it can be more easily reused in the future), and leave the JSON serialization for the code handling DuckDB query execution.